src/nwb2bids/_converters/_session_converter.py

```diff
@@ -55,7 +56,16 @@ def from_nwb_paths(
             if nwb_path.is_file():
                 all_nwbfile_paths.append(nwb_path)
             elif nwb_path.is_dir():
-                all_nwbfile_paths += list(nwb_path.rglob(pattern="*.nwb"))
+                all_nwbfile_paths += [
+                    path
+                    for path in nwb_path.rglob(pattern="*.nwb")
+                    if (
+                        # Ignore contents in hidden folders; e.g. .git, __pycache__, etc.
+                        not any(part.startswith(".") for part in path.parts)
+                        # Ignore DataLad files that are not yet retrieved from the annex
+                        and _content_is_retrieved(file_path=path)
```

Comment:
"explicit is better than implicit"

I think condition should be optional and not default really. For quick resolution -- you could make it secondary check and issue a warning to list those and continue.